src/main/java/org/elasticsearch/script/CompiledScript.java

```diff
@@ -24,20 +24,42 @@
  */
 public class CompiledScript {
 
+    private final ScriptService.ScriptType type;
+    private final String name;
     private final String lang;
     private final Object compiled;
 
     /**
      * Constructor for CompiledScript.
+     * @param type The type of script to be executed.
+     * @param name The name of the script to be executed.
      * @param lang The language of the script to be executed.
      * @param compiled The compiled script Object that is executable.
      */
-    public CompiledScript(String lang, Object compiled) {
+    public CompiledScript(ScriptService.ScriptType type, String name, String lang, Object compiled) {
```

Comment:
Would it make sense to move ScriptType out of ScriptService at some point (doesn't need to be in this PR necessarily)